### PR TITLE
[DEVELOPER 3401 & 3407] 2 column layout & compact article list assemblies

### PR DESCRIPTION
**Pull request by rogeralloymagnetic**
### JIRA Issue Link
* https://github.com/redhat-developer/developers.redhat.com/issues/3401
* https://github.com/redhat-developer/developers.redhat.com/issues/3407

### Verification Process
0. Login: https://developer-preview-3411.ext.us-west.dc.preprod.paas.redhat.com/user/login
1. Add "Layout: Two Column with Left Sidebar" assembly to homepage https://developer-preview-3411.ext.us-west.dc.preprod.paas.redhat.com/home/edit
2. Add "Compact Dynamic Article List" assembly to column 1
3. Add any hero assembly to the first slot in column 2
4. Add any content assembly to the second slot in column 2

### Known Issues
1. `compact_dynamic_articles_list` is not pulling in comment count

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend/src/styles/rhd-theme/components/_rhd-c-two-col-hero-layout.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                }
+
+            }
+
+            .field--name-field-column-2 {
+                > .assembly.component {
+                  // Remove margin-top from first assembly rendered in 2nd
+                  // column within the Layout 2 col, sidebar left assembly type.
+                  // This makes it match the behavior of the left, 1st column.
+                  @media screen and (max-width: $pf-global--breakpoint--lg) {
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend/src/styles/rhd-theme/components/_rhd-c-two-col-hero-layout.scss]: I accidentally used the wrong media query logic here. I wanted the opposite and didn't notice at the time of pushing the last commit. I will fix this now.

**Comment by jordanpagewhite** [comment]
@sjcox-rh and @joeschram This will be ready to review once the PR environment rebuilds. I already reviewed it, but I made a tiny change to some media query logic controlling the spacing of the first element in the 2nd column. There will now be now margin top applied to that component, specifically, for desktop viewports so that it aligns properly with the first column. The margin-top will still be applied for viewports under desktop so that there is space between components/assemblies.

**Comment by joeschram** [comment]
@jordanpagewhite & @sjcox-rh I did a test of the above, can show that the new assembly worked as expected. With this new assembly added, I was able to remove the current Dynamic Article List w/ Comments further down the page and instead front a row of Katacoda courses (see screenshot.)

Some things to note:

1. No Disqus comments showed up on the new Compact Dynamic Article List (I'm assuming this is b/c we are in a PR env.)
2. When testing responsiveness, I saw the Curated Content cards (and enclosed images) scale/expand to fill the viewport—**we'll need to make sure the image assets used on these cards are designed and sized properly to look good across various widths. The sample images I put in here are not what we should use for real.** 
3. I did not build the new Rich Text assembly to promote OpenShift 4.2 katacodas (as spec'd in [the doc](https://docs.google.com/document/d/1omSrY4WzW8fpRlOJ_Vl2a4ZeCau0ANsRIS7u-lpsyPc/edit)) because I did not have the new image. If we can keep this PR environment live after these changes are merged, @sjcox-rh you will be able to just open production Drupal in one browser and load this PR environment in another, and copy-paste the other text content. Note above: the Curated Content images needed to be replaced.

20191201 Homepage Test

**Comment by joeschram** [comment]
[Support for Home Page](https://trello.com/c/xpBkmUPs/726-support-for-home-page)

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robcoffman** [comment]
rebuild this please

**Comment by robcoffman** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by rogeralloymagnetic** [comment]
@sjcox-rh @joeschram There is a section in the description with the three known issues when I put this ticket down last week. 1 & 2 in that list cover two of the issues you guys are mentioning.

@jordanpagewhite were you able to merge in the changes from #1 in the known issues list? Also, were you able to address 2 & 3 or do those still need attention this week?

**Comment by sjcox-rh** [comment]
@rogeralloymagnetic could you possibly address the changes I requested yesterday regarding spacing/padding today? The updated home page has to go live today and Im hesitant to have this merge as is with the current padding limitations.

**Comment by alistairmc** [comment]
@rogeralloymagnetic has this had master merged into it recently?

**Comment by rogeralloymagnetic** [comment]
@alistairmc Probably not, but I'll be working on it this morning.

**Comment by alistairmc** [comment]
@rogeralloymagnetic I added a fix for the padding but unfortunately there is some more css causing an issue 

in _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/css/rhdp2.css

the following needs to be removed however, this could have a wider impact. This was not highlighted in the frontend as that css only exists in drupal

/* This is necessary to have consistent column width with nested components */
.component .component {
  padding: 16px 0;
}

**Comment by sjcox-rh** [comment]
@rogeralloymagnetic @jordanpagewhite 

I think this is good to merge as is!

Could we open up a follow up issue to fix the outstanding items for this assembly:

1. Comment Count
2. Update left and right padding for all hero and rich text assemblies to be 24px when used in this nested assembly. Is this possible to do? I realize that the left right padding is being pulled from default styles of these assemblies but is it possible to override when used in this specific assembly?
3. For curated content, when using an odd number of cards, the card fills page width at the medium breakpoint. Could we possibly keep it so it retains the two column card layout?
Screen Shot 2019-12-06 at 10 22 00 AM

Issues I noticed. @rogeralloymagnetic I know you said you noticed some others so feel free to add them to the follow up issue.

**Comment by rogeralloymagnetic** [comment]
@sjcox-rh in reply to your last comment:
1- I've created a new ticket for this here: https://github.com/redhat-developer/developers.redhat.com/issues/3434
2- Yes, we can override per layout. I'll update this today.
3- I'll also add this today. Let me know if there's any specific design you want besides the general assembly CTA button.
4- This is probably best to hand over to FE.

**Comment by sjcox-rh** [comment]
@rogeralloymagnetic 

Thanks for working on these changes. For the view more articles cta for column 1, let's use the medium CTA (so light blue outline button).

And ill create a FE ticket for the card width issue I called out.  Thanks!

**Comment by robpblake** [comment]
rebuild this please

**Comment by sjcox-rh** [comment]
@rogeralloymagnetic @jordanpagewhite is this able to get merged since we have follow up tickets created?

**Comment by jordanpagewhite** [comment]
@sjcox-rh Yes, I will try to get to this earlier this afternoon.

**Comment by jordanpagewhite** [comment]
This Managed Platform Build is being marked as a failure, but it seems to be fine in Jenkins and when visiting the PR environment URL: https://developer-preview-3411.ext.us-west.dc.preprod.paas.redhat.com/ . I am not sure why The Jenkins integration here in Github thinks it failed.

**Comment by robpblake** [comment]
retest this please

**Pull request merged**